File: Chapter 03/Python/3.2 Coupling Exercise/After/user_data_service.py
from user_report_data import UserReportData

class UserDataService:
    def get_user_report(self, user_id):
        print("    [Service] Building chunky report payload internally...")
        return UserReportData(name="Jane Doe", email="jane.doe@example.com", total_spent=199.90)

File: Chapter 03/C#/3.2 Coupling Exercise/After/UserDataService.cs
using System;

namespace Chapter03.CouplingTest.After;

public class UserDataService
{
    public UserReportData GetUserReport(int userId) 
    { 
        Console.WriteLine("    [Service] Building chunky report payload internally...");
        
        // The service orchestrates its own data now
        return new UserReportData 
        { 
            Name = "Jane Doe",
            Email = "jane.doe@example.com",
            TotalSpent = 199.90m // (99.95 * 2 orders)
        };
    }
}